app/controllers/questions_controller.rb

```diff
@@ -0,0 +1,94 @@
+class QuestionsController < ApplicationController
+  def index
+    @questions = Question.most_recent
+  end
+
+  def popular
+    @questions = Question.most_votes
+    render 'index'
+  end
+
+  def show
+    @question = Question.find_by(id: params[:id])
+    @answer = Answer.new
+    @answers = @question.sort_answers
+  end
+
+  def new
+    @question = Question.new
+    @tag = Tag.new
+  end
+
+  def create
+    @question = Question.new(question_params)
+    if @question.save
+    tag_params[:name].split(' ').each do |tag_name|
+      tag = Tag.find_or_create_by(name: tag_name)
+      @question.tags << tag
+    end
+      flash.notice = "Your question was successfully created."
+      redirect_to question_path(@question)
+    else
+      flash[:error] = "Your question was not successfully created. Make sure all fields are filled in."
```

Comment:
You need flash.now here - render isn't a new request so this error would actually show on the following page not the re-render of new. flash.now will place the flash message in the current request rather than the next one.